curriculum/challenges/english/25-front-end-development/quiz-dynamic-programming/66edd5ffe490bef99ea99395.md

```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is memoization in dynamic programming?
 
 #### --distractors--
 
-Placeholder distractor 1
+Memoization is a technique for compressing data to save memory.
 
 ---
 
-Placeholder distractor 2
+Memoization is a method for sorting data in ascending order
 
 ---
 
-Placeholder distractor 3
+Memoization is a process of encrypting data for security purposes.
 
 #### --answer--
 
-Placeholder answer
+Memoization is a technique where you store the results of expensive function calls and reuse them when the same inputs occur again.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How does dynamic programming differ from divide and conquer?
 
 #### --distractors--
 
-Placeholder distractor 1
+Dynamic programming always uses a greedy approach to solve problems
 
 ---
 
-Placeholder distractor 2
+Divide and conquer algorithms do not use recursion.
 
 ---
 
-Placeholder distractor 3
+Dynamic programming is only used for sorting algorithms
 
 #### --answer--
 
-Placeholder answer
+Dynamic programming solves each subproblem only once and stores the results, while divide and conquer may solve the same subproblem multiple times.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the time complexity of the Fibonacci sequence using dynamic programming? 
 
 #### --distractors--
 
-Placeholder distractor 1
+O(n^2)
 
 ---
 
-Placeholder distractor 2
+O(log n)
 
 ---
 
-Placeholder distractor 3
+O(2^n)
 
 #### --answer--
 
-Placeholder answer
+O(n)
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is a greedy algorithm?
 
 #### --distractors--
 
-Placeholder distractor 1
+A greedy algorithm always finds the globally optimal solution.
 
 ---
 
-Placeholder distractor 2
+A greedy algorithm uses backtracking to find the best solution.
 
 ---
 
-Placeholder distractor 3
+A greedy algorithm is only used for sorting problems
 
 #### --answer--
 
-Placeholder answer
+A greedy algorithm is an approach for solving problems by making the locally optimal choice at each stage with the hope of finding a global optimum
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the difference between dynamic programming and greedy algorithms?
 
 #### --distractors--
 
-Placeholder distractor 1
+Dynamic programming always makes the locally optimal choice at each step
 
 ---
 
-Placeholder distractor 2
+Greedy algorithms store solutions to subproblems to avoid redundant calculations
 
 ---
 
-Placeholder distractor 3
+Dynamic programming is only used for problems without overlapping subproblems.
 
 #### --answer--
 
-Placeholder answer
+Dynamic programming solves problems by combining solutions to subproblems and storing results, while greedy algorithms make a series of choices, each of which looks best at the moment.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Give an example of a problem that can be solved using a greedy algorithm.
```

Comment:
I am confused about the question. Isn't this an example of a greedy algorithm for the coin change problem?

```

 The greedy algorithm for making change would work as follows: 


     Start with the largest coin value that is less than or equal to the amount to be changed. In this case, the largest coin less than  36  is  20  . 

     Subtract the largest coin value from the amount to be changed, and add the coin to the solution. In this case, subtracting  20  from  36  gives  16  , and we add a  20  coin to the solution. 

     Repeat steps 1 and 2 until the amount to be changed becomes 0. 


 So, using the greedy algorithm, the solution for making change for 36 would be one 20 coins, one 10 coin, one 5 coins and one 1 coin needed. 
``` 

From : https://www.geeksforgeeks.org/introduction-to-greedy-algorithm-data-structures-and-algorithm-tutorials/